spring-kafka/src/main/java/org/springframework/kafka/listener/ListenerInvokingBatchErrorHandler.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.kafka.listener;
+
+import java.util.function.Supplier;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+
+/**
+ * A batch error handler that is capable of invoking the listener during error handling.
+ *
+ * @author Gary Russell
+ * @since 2.3.7
+ *
+ */
+public interface ListenerInvokingBatchErrorHandler extends ContainerAwareBatchErrorHandler {
```

Comment:
Why do we need this abstraction if we have the same callback on the `BatchErrorHandler` or `ContainerAwareBatchErrorHandler`?

Reply:
So that implementations only have to implement this method.